### PR TITLE
Fixes #29128 - Fix Style/Lambda cop

### PR DESCRIPTION
**Pull request by mmoll**

### CODE REVIEW

**Comment by mmoll** [comment]
We do have this one enabled in core and it's an easy auto-correct.

**Comment by ekohl** [comment]
I'm not sure about this one. Is this purely a style thing? Are there differences?

**Comment by mmoll** [comment]
AFAICT, there should be no difference: http://rubyforadmins.com/lambda-functions

The cop enforces the arrow notation specifially only for one-liners.

**Comment by lzap** [comment]
Honestly I see zero value in this one. Actually I like `lambda` syntax more easy to read, that's probably just me.

**Comment by mmoll** [comment]
> Honestly I see zero value in this one.

If we wouldn't have this in core since many years, I'd just leave it disabled. The alternative would be to disable it in core.

File: test/root/root_v2_api_test.rb [hunk 1]
```diff
@@ -25,8 +25,8 @@ def app
   end
 
   def test_features
-    proc2 = lambda{ ['a', 'b']}
-    proc3 = lambda{raise "Should not be called"}
+    proc2 = -> { ['a', 'b'] }
+    proc3 = -> {raise "Should not be called" }
```
Comment (ekohl) [inline test/root/root_v2_api_test.rb]: This whitespacing looks inconsistent.

Comment (mmoll) [reply test/root/root_v2_api_test.rb]: aw, I fixed up the `proc2` line manually. Now also `proc3`.

Comment (mmoll) [reply test/root/root_v2_api_test.rb]: I guess, there would be another (currently disabled) cop also for that.

**Review by ekohl — APPROVED** [review]

**Pull request merged**